### PR TITLE
Faster date parsing

### PR DESCRIPTION
**Pull request by JHolba**
datetime parsing while unpacking json in cloudevents is *very* slow currently.

**Approach**
Replaced datetime parsing library with a new one that is included in python 3.11. Using backport package for older python versions. The new one should be a bit more than two orders of magnitude faster.

- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [x] Make sure tests pass locally (after every commit!)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

**Comment by JHolba** [comment]
the backports package is written by the author of the ciso8601 python package. ciso8601 may be a better parser, but I figured it is better to go for a package that we can phase out when we move to python 3.11. 
https://github.com/closeio/ciso8601

File: src/ert/serialization/_evaluator.py [hunk 1]
```diff
@@ -1,13 +1,15 @@
-import datetime
 import json
+from datetime import date, datetime
 from typing import Any
 
-from dateutil import parser
+from backports.datetime_fromisoformat import MonkeyPatch  # type: ignore
+
+MonkeyPatch.patch_fromisoformat()
```
Comment (jonathan-eq) [inline src/ert/serialization/_evaluator.py:7]: Are those the only places we do date parsing?

**Review by jonathan-eq — APPROVED** [review]
Very good!

File: src/ert/serialization/_evaluator.py (hunk 1 above)
Comment (JHolba) [reply src/ert/serialization/_evaluator.py:7]: No, but it was the only places that mattered according to the profiling I did

**Pull request merged**